src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -276,5 +276,11 @@ public override int GetHashCode(string obj)
         // Equals/GetHashCode methods for the comparer itself. 
         public override bool Equals(object obj) => obj is OrdinalIgnoreCaseComparer;
         public override int GetHashCode() => nameof(OrdinalIgnoreCaseComparer).GetHashCode();
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.SetType(typeof(OrdinalComparer));
+            info.AddValue("_ignoreCase", true);
```

Comment:
- Make OrdinalComparer to be exactly like what it used to be in desktop.
- Inherit OrdinalCaseSensitiveComparer and OrdinalIgnoreCaseComparer that have the optimized implementations
   - In OrdinalCaseSensitiveComparer.GetObjectData and OrdinalIgnoreCaseComparer.GetObjectData, do info.SetType(typeof(OrdinalComparer)) so that they serialize into know desktop type

Ssame pattern as what is used for EnumEqualityComparer. It also has more optimized version and serializes as the less optimized version.